Libraries/Opc.Ua.Configuration/ApplicationInstance.cs

```diff
@@ -460,7 +526,7 @@ public async Task DeleteApplicationInstanceCertificate(CancellationToken ct = de
 
             // reload the certificate from disk in the cache.
             var passwordProvider = configuration.SecurityConfiguration.CertificatePasswordProvider;
-            await configuration.SecurityConfiguration.ApplicationCertificate.LoadPrivateKeyEx(passwordProvider).ConfigureAwait(false);
+            await id.LoadPrivateKeyEx(passwordProvider).ConfigureAwait(false);
```

Comment:
the password provider can handle multiple app certs and returns the right pw for each?

Reply:
Isn't it's interface designed to do so ?  The GetPassword(CertificateIdentifier) allows it.